src/riscv/lib/src/machine_state/memory.rs

```diff
@@ -307,6 +317,33 @@ pub trait MemoryConfig: Send + 'static {
         F: FnManager<Ref<'a, M>>;
 }
 
+/// Trait that handles memory permission updates for the jit.
+/// The jit router needs to know when the memory permissions
+/// are updated so it can invalidate the jit instances.
+pub trait MemoryGovernanceListener {
+    fn handle_memory_permission_change(
```

Comment:
This misses a documentation comment.

```suggestion
    /// Called on changes to memory permissions.
    fn handle_memory_permission_change(
```